WordPress/wporg-main-2022
# Try parsing block content and replacing strings with i18n functions

See #15 — This updates the `build:patterns` script to inject i18n functions into the patterns, which can then be synced into GlotPress using the regular code translation process.

This uses the parsers from the pattern directory project to extract strings from the post/pattern content, then wraps them in `<?php _e( '%s', 'wporg' ); ?>`. The parsers themselves have not been updated, but the main parser class (`BlockParser` here, `parser.php`) has been updated to add the `replace_with_i18n` function. I also had to tweak `/parsers/Button.php` to support having a shortcode in `href`.

It would probably be better if these files were not duplicated, at least the `parsers/*`, but I'm not sure what the best approach would be.

The `export-content` script has been updated so it now needs to be called via wp-cli, so that we can use the block parsing & serialization functions. `build:patterns` handles that behind the scenes. The only configuration allowed now is the manifest file.

### Screenshots

There should be no visual change.

### How to test the changes in this Pull Request:

1. Run `yarn build:patterns`
2. The pattern files should not output text directly, they should use PHP i18n functions.
